client/src/Services/ApiService.js

```diff
@@ -1 +1,56 @@
-//API call services will come here
\ No newline at end of file
+//API call services will come here
+
+import axios from "axios";
+import { BASE_URL } from "../Utils/Urls";
+
+export const webService = async (url, type, options) => {
+  const baseUrl = BASE_URL;
+  const config = {
+    headers: { "Access-Control-Allow-Origin": "*" },
+  };
+  switch (type) {
+    case "get":
+      return await axios
+        .get(`${baseUrl}${url}`,config)
+        .then((res) => {
+          return res.data;
+        })
+        .catch((error) => {
+          return error;
+        });
```

Comment:
I understand trying to abstract out the api fetching. But I think we're mixing two different patterns and that has the potential to let errors slip through. 

Does the `.catch` throw a new Error so that the UI can react with that error? As I'm reading it we fetch from the API, the API returns 500 and `webService` returns that 500. The listener to `webService` has to do further checking to see if there was a 500 returned. 

I think I would understand this better if there was some testing around it.

Reply:
We are planning to create a common function to handle the errors. Usually, status codes are checked at the backend and the frontend receives a boolean value indicating whether there is an error and the error message. We'll be defining the function inside ApiService.js file itself. As of now, we dont have clarity on how the error response is being received.

Reply:
Yeah but that doesn't answer the question right?

You're using 2 different and in my opinion incompatible patterns here. 
 
Theres the `async / await` with a `try ... catch` pattern and there's the `then(). catch()` pattern. This is using both. Why?

Reply:
Thank you for pointing that out. I will remove `async / await` pattern from the function.